helix-term/src/commands.rs

```diff
@@ -5896,16 +5903,44 @@ fn shell(cx: &mut compositor::Context, cmd: &str, behavior: &ShellBehavior) {
         changes.push((from, to, Some(output)));
     }
 
-    if behavior != &ShellBehavior::Ignore {
-        let transaction = Transaction::change(doc.text(), changes.into_iter())
-            .with_selection(Selection::new(ranges, selection.primary_index()));
-        doc.apply(&transaction, view.id);
-        doc.append_changes_to_history(view);
-    }
+    let mut commands = Vec::with_capacity(changes.len());
+    match behavior {
```

Comment:
Don't you need to extend also let (from, to, deleted_len) = match behavior to not use default ?
Also currently looks like if there would be two commands or command with argument it will cause error. And are you sure you would want to replace ignore behaviour with none ? 😅

Reply:
Apologies for my pool rust programming.

> Don't you need to extend also let (from, to, deleted_len) = match behavior to not use default ?

IIUC, the `pipe-execute` behavior should do nothing to the selected text like `Ignore` behavior, so I keep it in the previous commit. But anyway now I've extended it if this makes you better.

> Also currently looks like if there would be two commands or command with argument it will cause error. 

I'm not sure whether you mean shell command to run or helix command from shell output. The former keeps same to other pipe behavior. The latter seems work fine in demo video. If you are talking about executing multiple helix commands from a single shell output, then we need to determine a delimiter, perhaps `\n`?

> And are you sure you would want to replace ignore behaviour with none ? 😅

Do you mean the line 5907? Seems no issue here.

```rust
// original
    if behavior != &ShellBehavior::Ignore {
    ...

// pr
    match behavior {
        &ShellBehavior::Ignore => (),
        ...
```